src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubCommitDetail.java

```diff
@@ -0,0 +1,62 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import hudson.model.Actionable;
+import hudson.model.Run;
+import jenkins.model.details.Detail;
+import jenkins.model.details.DetailGroup;
+import jenkins.plugins.git.AbstractGitSCMSource;
+import jenkins.scm.api.SCMDetailGroup;
+import jenkins.scm.api.SCMRevision;
+import jenkins.scm.api.SCMRevisionAction;
+import jenkins.scm.api.SCMSource;
+
+public class GitHubCommitDetail extends Detail {
+    public GitHubCommitDetail(Actionable object) {
+        super(object);
+    }
+
+    public String getIconClassName() {
+        return "symbol-git-commit-outline plugin-ionicons-api";
+    }
+
+    @Override
+    public String getDisplayName() {
+        SCMRevisionAction scmRevisionAction = getObject().getAction(SCMRevisionAction.class);
+        SCMRevision revision = scmRevisionAction.getRevision();
+
+        if (revision instanceof AbstractGitSCMSource.SCMRevisionImpl abstractRevision) {
+            return abstractRevision.getHash().substring(0, 7);
+        } else if (revision instanceof PullRequestSCMRevision pullRequestSCMRevision) {
+            return pullRequestSCMRevision.getPullHash().substring(0, 7);
```

Comment:
this definitely needed? i.e the git hash is different?

I'm guessing this is if someone has used a `pr-merge` strategy? and github has created a merge commit of the target branch and the PR branch?